src/request/mod.rs

```diff
@@ -132,27 +132,31 @@ pub trait DeliverableRequest<'a>: Sized + 'a {
 
     fn get_all_headers(&self) -> HeaderMap {
         let mut additional_headers = HeaderMap::new();
-        additional_headers.extend(self.get_custom_headers());
+        additional_headers.extend(self.get_custom_headers().iter().map(|(k, v)| (k.clone(), v.clone())));
         additional_headers.extend(self.tracing_headers());
         #[cfg(feature = "auth0")]
         additional_headers.extend(self.get_auth0_headers());
         additional_headers
     }
 
     #[doc(hidden)]
-    fn get_body(&self) -> Vec<u8>;
+    fn get_request_type(&self) -> RequestType;
 
     #[doc(hidden)]
-    fn get_request_type(&self) -> RequestType;
+    fn into_body(self) -> Vec<u8>;
 
-    fn get_form(&self) -> PrimaBridgeResult<Option<reqwest::multipart::Form>> {
-        Ok(None)
+    #[doc(hidden)]
+    /// Returns `Ok(Err(Self))` if this request has no form data
+    fn into_form(self) -> PrimaBridgeResult<Result<reqwest::multipart::Form, Self>> {
```

Comment:
I don't know if i like return a result of a result where its error is Self.. Moreover, as above, i would call it something like `body_as_form` or something similar